askama_shared/src/generator.rs

```diff
@@ -353,11 +353,21 @@ impl<'a> Generator<'a> {
                     self.handle_include(state, ws, path);
                 },
                 Node::Call(ref ws, name, ref args) => self.write_call(state, ws, name, args),
-                Node::Macro(_, _) |
-                Node::Import(_, _) |
+                Node::Macro(_, ref m) => {
+                    if let AstLevel::Nested = level {
+                        panic!("macro blocks only allowed at the top level");
+                    }
+                    self.write_macro(m);
+                },
+                Node::Import(ref ws, _) => {
+                    if let AstLevel::Nested = level {
+                        panic!("import blocks only allowed at the top level");
+                    }
+                    self.write_import(ws);
+                },
                 Node::Extends(_) => {
```

Comment:
Should we also deal with the whitespace from the `Extends` nodes, to get it really right? (It doesn't currently parse a `WS`, but maybe it should to be totally consistent.)

Reply:
I don't think whitespace handling is applicable to ```Extends``` because the only content that is used from a child template is from inside blocks. I did some tests and from what I can see whitespace in the child template outside of blocks don't have any impact on the generation. 

It might be a good idea though to support ```{%-``` and ```-%}``` in the parser to standardize the syntax, even though it doesn't have any impact

Reply:
Ha, good point! Okay, then we can at least ignore it for the purposes of this PR. Or, maybe add a comment explaining this?